app/src/main/kotlin/com/intive/picover/auth/repository/AuthRepository.kt

```diff
@@ -41,15 +41,14 @@ class AuthRepository @Inject constructor(
 		firebaseAuth.signOut()
 	}
 
-	suspend fun userProfile(): Profile {
-		val photoUrl = userAvatarReference.downloadUrl.await()
-		return requireUser().let {
-			Profile(
-				photo = photoUrl,
-				name = it.displayName!!,
-				email = it.email!!,
-			)
-		}
+	fun fetchUserProfile(onSuccess: (profile: Profile) -> Unit, onFailure: () -> Unit) {
```

Comment:
You should expose flow, etc. here as a returned type instead using of old-fashioned java callback style

Reply:
I had a solution with `callbackFlow` for all this, but later I started reading that `flow` is unnecessary when we want to get a value once and we don't observe anything so I changed it to what it is

Reply:
Could you link the source?

Reply:
I mean the first sentence from flow documentation "In coroutines, a flow is a type that can emit multiple values sequentially, as opposed to suspend functions that return only a single value."

Reply:
Understood, then we should rather stick to the previous approach. Was there any particular reason to do that in a callback way?

Reply:
I had to add listeners to Firebase calls to handle it properly to fix this bug. I don't want to create listeners in ViewModel to update the state in its callback to not have Firebase classes in ViewModel. Thus I pass callbacks to the repository this way as I didn't find a better solution for this case.

Reply:
Maybe wrapping a suspend function with `kotlin.Result` as a returned type would help?

Reply:
@MarekMacko please take a look at my try with `suspendCoroutine`